src/libfuncs.rs

```diff
@@ -507,3 +507,258 @@ pub fn increment_builtin_counter<'ctx: 'a, 'a>(
         .result(0)?
         .into())
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use melior::dialect::DialectRegistry;
+    use melior::ir::Type;
+    use melior::utility::register_all_dialects;
+
+    /// Load all available MLIR dialects into the given context.
+    ///
+    /// This function initializes a new `DialectRegistry`, registers all available dialects
+    /// into it, appends the registry to the provided context, and then loads all available
+    /// dialects into the context.
+    pub fn load_all_dialects(context: &Context) {
```

Comment:
Please use the NativeContext struct, it handles all of this, the dialect loading, etc.

Reply:
Fixed